src/main/resources/com/ncc/aif/restricted_aif.shacl

```diff
@@ -0,0 +1,159 @@
+            }
+            GROUP BY $this
+            HAVING (COUNT(DISTINCT ?source) > 1)
+        """ ;
+    ] .
+
+#########################
+# Each edge justification is limited to two or fewer spans
+#------------------------
+# Can have a maximum of 2 justifiedBy/containedJustification properties
+aida:EdgeJustificationLimit
+    a sh:PropertyShape ;
+    sh:path ( aida:justifiedBy aida:containedJustification ) ;
+    sh:maxCount 2 ;
+    sh:message "Only 2 contained justifications allowed for an edge" .
+
+# Turn off CompoundJustificationMinimum globally
+aida:CompoundJustificationMinimum sh:deactivated true .
+
+#########################
+# Each edge justification must be represented uniformly in AIF by
+# aida:CompoundJustification, even if only one span is provided
+# Edges are assumed to be relation and event arguments
+#------------------------
+# Define restriction where justification can only be compound justification
+aida:EdgeJustificationCompound
+    a sh:PropertyShape ;
+    sh:path aida:justifiedBy ;
+    sh:class aida:CompoundJustification ;
+    sh:message "Edge justification must be of type aida:CompoundJustification" .
+
+# Enforce edge justification restriction on event and relation arguments
+aida:EventArgumentShape
+    sh:property aida:EdgeJustificationCompound, aida:EdgeJustificationLimit .
+aida:RelationArgumentShape
+    sh:property aida:EdgeJustificationCompound, aida:EdgeJustificationLimit .
+
+#########################
+# Video must use aida:KeyFrameVideoJustification. Remove ShotVideoJustification
+#------------------------
+# Restrict normal Justification types globally
+aida:JustificationPropertyShape
+    sh:xone (
+        [sh:class aida:TextJustification]
+        [sh:class aida:ImageJustification]
+        [sh:class aida:AudioJustification]
+        [sh:class aida:KeyFrameVideoJustification]
+        [sh:class aida:CompoundJustification] ) .
+
+# Restrict CompoundJustificationTypes globally
+aida:CompoundJustificationTypes
+    sh:xone (
+        [sh:class aida:TextJustification]
+        [sh:class aida:ImageJustification]
+        [sh:class aida:AudioJustification]
+        [sh:class aida:KeyFrameVideoJustification] ) .
+
+#########################
+# Members of clusters are entity objects, relation objects, and event objects (not clusters)
+#------------------------
+aida:ClusterMembershipShape
+    sh:property [
+        a sh:PropertyShape ;
+        sh:path aida:clusterMember ;
+        sh:xone (
+            [sh:class aida:Entity]
+            [sh:class aida:Event]
+            [sh:class aida:Relation]
+        ) ;
+        sh:message "Cluster member type not allowed to be part of cluster"
+    ] .
+
+#########################
+# Entity, Relation, and Event object is required to be part of at least one cluster.
+# This is true even if there is nothing else in the cluster
+#------------------------
+aida:EntityShape
```

Comment:
Good question! For both this comment and the one below, the name has to be aida:EntityShape and aida:EventShape because we are not actually defining full shapes here. We are simply adding properties to pre-existing shapes (ie., aida:EntityShape and aida:EventShape, respectively).

Reply:
Ah okay.